[build] failed to build on Ubuntu 17.10

Configuration:
Ubuntu 17.10
MLNX_OFED_LINUX-4.4-1.0.0.0
UCX version (latest master):
```sh
$ git show HEAD
commit 830fcec694b88d527d58cd27d0f862f4bcfd2aaa (HEAD -> master, origin/master, origin/HEAD)
Merge: c60c4735 c61d6806
Author: Yossi Itigin <yosefe@mellanox.com>
Date:   Mon Nov 5 22:35:15 2018 +0200

    Merge pull request #3021 from Artemy-Mellanox/topic/fix-av-2

    UCT/IB: Fix force global AV
```

Reproducation:
```sh
$ ./autogen.sh
$ ./contrib/configure-release
$ make -j1
```

Output:
```sh
make[2]: Entering directory '/auto/mtrswgwork/andreyma/hpcx/ucx/src/uct'
  CC       ib/mlx5/libuct_la-ib_mlx5_log.lo
In file included from ib/mlx5/ib_mlx5_log.h:10:0,
                 from ib/mlx5/ib_mlx5_log.c:7:
ib/mlx5/ib_mlx5.h:90:8: error: redefinition of ‘struct mlx5_grh_av’
 struct mlx5_grh_av {
        ^~~~~~~~~~~
In file included from ib/mlx5/ib_mlx5.h:37:0,
                 from ib/mlx5/ib_mlx5_log.h:10,
                 from ib/mlx5/ib_mlx5_log.c:7:
/usr/include/infiniband/mlx5dv.h:447:8: note: originally defined here
 struct mlx5_grh_av {
        ^~~~~~~~~~~
In file included from /usr/include/arpa/inet.h:22:0,
                 from /labhome/andreyma/workspace/hpcx/ucx/src/ucs/sys/sys.h:33,
                 from /labhome/andreyma/workspace/hpcx/ucx/src/uct/base/uct_iface.h:18,
                 from /labhome/andreyma/workspace/hpcx/ucx/src/uct/base/uct_md.h:10,
                 from /labhome/andreyma/workspace/hpcx/ucx/src/uct/ib/base/ib_md.h:14,
                 from /labhome/andreyma/workspace/hpcx/ucx/src/uct/ib/base/ib_iface.h:10,
                 from /labhome/andreyma/workspace/hpcx/ucx/src/uct/ib/base/ib_log.h:11,
                 from ib/mlx5/ib_mlx5.h:13,
                 from ib/mlx5/ib_mlx5_log.h:10,
                 from ib/mlx5/ib_mlx5_log.c:7:
ib/mlx5/ib_mlx5_log.c: In function ‘uct_ib_mlx5_dump_dgram’:
ib/mlx5/ib_mlx5_log.c:191:27: error: ‘struct mlx5_wqe_av’ has no member named ‘dqp_dct’
              ntohl(base_av->dqp_dct & ~UCT_IB_MLX5_EXTENDED_UD_AV));
                           ^
In file included from /usr/include/x86_64-linux-gnu/bits/byteswap.h:35:0,
                 from /usr/include/endian.h:60,
                 from /usr/include/x86_64-linux-gnu/sys/types.h:194,
                 from /labhome/andreyma/workspace/hpcx/ucx/src/uct/api/uct_def.h:14,
                 from /labhome/andreyma/workspace/hpcx/ucx/src/uct/api/uct.h:12,
                 from /labhome/andreyma/workspace/hpcx/ucx/src/uct/base/uct_worker.h:10,
                 from ib/mlx5/ib_mlx5.h:12,
                 from ib/mlx5/ib_mlx5_log.h:10,
                 from ib/mlx5/ib_mlx5_log.c:7:
ib/mlx5/ib_mlx5_log.c:195:56: error: ‘struct mlx5_wqe_av’ has no member named ‘rlid’
         snprintf(p, endp - p, " rlid %d", ntohs(base_av->rlid));
                                                        ^
ib/mlx5/ib_mlx5_log.c:199:33: error: ‘struct mlx5_wqe_av’ has no member named ‘dqp_dct’
     if (mlx5_av_base(&dgseg->av)->dqp_dct & UCT_IB_MLX5_EXTENDED_UD_AV) {
                                 ^~
In file included from /labhome/andreyma/workspace/hpcx/ucx/src/ucs/type/status.h:12:0,
                 from /labhome/andreyma/workspace/hpcx/ucx/src/uct/api/uct_def.h:10,
                 from /labhome/andreyma/workspace/hpcx/ucx/src/uct/api/uct.h:12,
                 from /labhome/andreyma/workspace/hpcx/ucx/src/uct/base/uct_worker.h:10,
                 from ib/mlx5/ib_mlx5.h:12,
                 from ib/mlx5/ib_mlx5_log.h:10,
                 from ib/mlx5/ib_mlx5_log.c:7:
/labhome/andreyma/workspace/hpcx/ucx/src/ucs/sys/compiler_def.h:107:35: error: ‘struct mlx5_wqe_av’ has no member named ‘reserved0’
     ((unsigned long)&( ((_type*)0)->_member ))
                                   ^
ib/mlx5/ib_mlx5.h:84:38: note: in expansion of macro ‘ucs_offsetof’
                                      ucs_offsetof(struct mlx5_wqe_av, reserved0[0])))
                                      ^~~~~~~~~~~~
ib/mlx5/ib_mlx5_log.c:200:18: note: in expansion of macro ‘mlx5_av_grh’
         grh_av = mlx5_av_grh(&dgseg->av);
                  ^~~~~~~~~~~
/labhome/andreyma/workspace/hpcx/ucx/src/ucs/sys/compiler_def.h:107:35: error: ‘struct mlx5_wqe_av’ has no member named ‘reserved0’
     ((unsigned long)&( ((_type*)0)->_member ))
                                   ^
ib/mlx5/ib_mlx5.h:85:37: note: in expansion of macro ‘ucs_offsetof’
 #  define UCT_IB_MLX5_AV_BASE_SIZE  ucs_offsetof(struct mlx5_wqe_av, reserved0[0])
                                     ^~~~~~~~~~~~
ib/mlx5/ib_mlx5_log.c:217:16: note: in expansion of macro ‘UCT_IB_MLX5_AV_BASE_SIZE’
         return UCT_IB_MLX5_AV_BASE_SIZE;
                ^~~~~~~~~~~~~~~~~~~~~~~~
ib/mlx5/ib_mlx5_log.c:219:1: error: control reaches end of non-void function [-Werror=return-type]
 }
 ^
cc1: all warnings being treated as errors
Makefile:1519: recipe for target 'ib/mlx5/libuct_la-ib_mlx5_log.lo' failed
make[2]: *** [ib/mlx5/libuct_la-ib_mlx5_log.lo] Error 1
make[2]: Leaving directory '/auto/mtrswgwork/andreyma/hpcx/ucx/src/uct'
Makefile:687: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/auto/mtrswgwork/andreyma/hpcx/ucx'
Makefile:557: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/config/m4/ib.m4
+++ b/config/m4/ib.m4
@@ -204,7 +204,7 @@ AS_IF([test "x$with_ib" == xyes],
                        AC_CHECK_HEADERS([infiniband/mlx5dv.h],
                                [with_mlx5_hw=yes
                                 with_mlx5_dv=yes
-                                mlx5_include=mlx5dv.h])])
+                                mlx5_include=mlx5dv.h], [], [ ])])
 
               AS_IF([test "x$with_mlx5_dv" == xyes -a "x$have_cq_io" == xyes ], [
                        AC_CHECK_DECLS([

--- a/src/uct/ib/mlx5/ib_mlx5.h
+++ b/src/uct/ib/mlx5/ib_mlx5.h
@@ -33,7 +33,6 @@
 #endif
 
 #if HAVE_INFINIBAND_MLX5DV_H
-#  include <infiniband/verbs.h>
 #  include <infiniband/mlx5dv.h>
 #else
 #  include <infiniband/mlx5_hw.h>

